public/app/features/plugins/admin/hooks/usePlugins.tsx

```diff
@@ -58,22 +58,48 @@ type FilteredPluginsState = {
   plugins: CatalogPlugin[];
 };
 
-export const usePluginsByFilter = (searchBy: string, filterBy: string): FilteredPluginsState => {
-  const { loading, error, plugins } = usePlugins();
+type PluginsByFilterType = {
+  searchBy: string;
+  filterBy: string;
+  filterByType: string;
+};
+
+const filters = {
+  filterBy: (plugin: CatalogPlugin, filterBy?: string) =>
+    filterBy === 'installed' ? plugin.isInstalled : !plugin.isCore,
+
+  filterByType: (plugin: CatalogPlugin, filterByType?: string) =>
+    filterByType === 'all' || plugin.type === filterByType,
+
+  searchBy: (plugin: CatalogPlugin, searchBy?: string) => {
+    if (!searchBy) {
+      return true;
+    }
+    const fields: String[] = [];
+    if (plugin.name) {
+      fields.push(plugin.name.toLowerCase());
+    }
+
+    if (plugin.orgName) {
+      fields.push(plugin.orgName.toLowerCase());
+    }
 
-  const installed = useMemo(() => plugins.filter((plugin) => plugin.isInstalled), [plugins]);
+    return fields.some((f) => f.includes(searchBy.toLowerCase()));
+  },
+};
```

Comment:
**Nitpick:** the name of the filter methods doesn't fully align with what they are doing

I understand that the names of these keys have been chosen to make the user-facing API (URL query params) cleaner and easy to understand, which I really like. On the other hand I find the functions not fully covering what they they are doing, e.g. `filterBy` is actually checking if the plugin is installed, although I would expect it being something more generic based on its name.

I am not 100% sure that what I am proposing is any better, so only leaving it here as an idea:

```javascript
type PluginFilter = (plugin: CatalogPlugin, query: string) => boolean;

// Hide core plugins unless explicitly filtering for installed ones
const isInstalled: PluginFilter = (plugin, query) =>  query === 'installed' ? plugin.isInstalled : !plugin.isCore,

const hasType: PluginFilter = (plugin, query) => filterByType === 'all' || plugin.type === filterByType;

const doesMatchKeyword = (plugin, keyword) => {
    if (!searchBy) {
      return true;
    }
    const fields: String[] = [];
    if (plugin.name) {
      fields.push(plugin.name.toLowerCase());
    }

    if (plugin.orgName) {
      fields.push(plugin.orgName.toLowerCase());
    }

    return fields.some((f) => f.includes(searchBy.toLowerCase()));
}

const URLFilterHandlers = {
    filterBy: isInstalled,
    filterByType: hasType,
    searchBy: doesMatchKeyword
};
```

Reply:
Thanks, this is much nicer @leventebalogh. Have updated to match your suggestions. 🥳